portal/views/user.py

```diff
@@ -372,9 +372,9 @@ def access_url(user_id):
     if not has.isdisjoint(not_allowed):
         abort(400, "Access URL not provided for privileged accounts")
 
-    if ROLE.WRITE_ONLY not in has:
+    if set((ROLE.ACCESS_ON_VERIFY, ROLE.WRITE_ONLY)).isdisjoint(has):
         # KEEP this restriction.  Weak authentication (which the
-        # returned URL provides) should only be available for WRITE_ONLY users
+        # returned URL provides) should only be available for these roles
         abort(400, "Account {} lacks WRITE_ONLY role".format(user_id))
```

Comment:
Should this abort message reference the 'ACCESS_ON_VERIFY' role?

Reply:
yup!